### ISSUE TITLE
Problem with big files (200mb) in API inbox

### ISSUE THREAD

**Issue by BrOrlandi** [chatwoot/chatwoot#10470]
### Describe the bug

I’m using an API inbox integration connected to our company's WhatsApp Business account. Couple days ago, a client sent a large 200MB Photoshop (PSD) file, which caused our server to crash completely—I had to restart the machine to bring everything back online. I verified this was the issue because a team member attempted to send the same file again, and it caused another crash.

We're hosted on Oracle Cloud with a machine of 1 CPUs and 4GB of RAM, which usually handles Chatwoot v3.12 without any issues.

My question is: Can Chatwoot be prevented from downloading or processing files above a certain size limit? Currently, file uploads to S3 are enabled for Chatwoot, and it seems that Chatwoot might be downloading the file through the API integration before uploading to S3, causing the crash.  As it stands, any client sending a large file, such as 200MB, is enough to make my Chatwoot server go down. I'm using  S3 compatible Object Storage from Oracle Cloud also.

### To Reproduce

Environment:
1 CPU and 4GB of RAM
Chatwoot v3.12

Set up an API inbox.

Send a message from the customer to the inbox with a file of at least 200mb.

The server should crash trying to download and process this file. 

### Expected behavior

Chatwoot shoud not crash.

### Environment

app.chatwoot.com

### Cloud Provider

Other [please specify in the description]

### Platform

Browser

### Operating system

Linux

### Browser and version

Chrome (doesn't matter the problem happens on the server)

### Docker (if applicable)

docker version

Client: Docker Engine - Community
 Version:           27.3.1
 API version:       1.47
 Go version:        go1.22.7
 Git commit:        ce12230
 Built:             Fri Sep 20 11:40:59 2024
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          27.3.1
  API version:      1.47 (minimum version 1.24)
  Go version:       go1.22.7
  Git commit:       41ca978
  Built:            Fri Sep 20 11:40:59 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.7.22
  GitCommit:        7f7fdf5fed64eb6a7caf99b3e12efcf9d60e311c
 runc:
  Version:          1.1.14
  GitCommit:        v1.1.14-0-g2c9f560
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

Client: Docker Engine - Community
 Version:    27.3.1
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.17.1
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.29.7
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 8
  Running: 8
  Paused: 0
  Stopped: 0
 Images: 7
 Server Version: 27.3.1
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7f7fdf5fed64eb6a7caf99b3e12efcf9d60e311c
 runc version: v1.1.14-0-g2c9f560
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.8.0-1014-oracle
 Operating System: Ubuntu 24.04.1 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 3.819GiB
 Name: chatwoot
 ID: 52b6b1d7-d6fc-437f-8f58-1cb2cdae1956
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled

### Additional context

This issue was discussed before on Discord Community: https://discord.com/channels/647412545203994635/1302399701915533312

**Comment by pranavrajs** [comment]
@BrOrlandi Interesting! I don't think we support PSD files. How did you upload this to Chatwoot?

**Comment by shivamkb17** [comment]
I think you are using whatsapp API inbox using official API or you are using any third party app to connect to chatwoot using API?

**Comment by BrOrlandi** [comment]
Yes @shivamkb17, I'm using a third-party app to connect to Chatwoot via API, and the file was sent by a customer via WhatsApp.

**Comment by pranavrajs** [comment]
@BrOrlandi It’s interesting to note that the WhatsApp Cloud API documentation states the maximum file size for any format is 100MB. <br><br>To quote directly from the website: 

> The maximum supported file size for media messages on Cloud API is 100MB. In the event the customer sends a file that is greater than 100MB, you will receive a webhook with error code [131052](https://developers.facebook.com/docs/whatsapp/cloud-api/support/error-codes#other-errors) and `title`:<br><br>"*Media file size too big. Max file size we currently support: 100MB. Please communicate with your customer to send a media file that is smaller than 100MB"*

We have not handled this case. 

> Can Chatwoot be prevented from downloading or processing files above a certain size limit? 

At the moment, no. We will consider this in the upcoming updates. 

> Currently, file uploads to S3 are enabled for Chatwoot, and it seems that Chatwoot might be downloading the file through the API integration before uploading to S3, causing the crash. As it stands, any client sending a large file, such as 200MB, is enough to make my Chatwoot server go down. I'm using S3 compatible Object Storage from Oracle Cloud also.

This is possible, and the likely reason is that your machine is running out of memory. How much storage space is left on the device?

**Comment by pranavrajs** [comment]
> I'm using a third-party app to connect to Chatwoot via API

Just to confirm. Are you using un-official API? Or are you using Cloud API/self-hosted API via a BSP?

**Comment by BrOrlandi** [comment]
@pranavrajs Yes, I'm using an unofficial WhatsApp API. I'm using the Evolution API, a great project that connects Whatsapp directly to Chatwoot: https://github.com/EvolutionAPI/evolution-api/.

Besides the cloud API limits to 100mb on upload, the normal API seems to have no limit and it's possible to send a file of 200mb or even more. 

My device currently has 30 GB in available space, so it should not be an issue.

**Comment by shivamkb17** [comment]
ok, I understood the issue, This issue is due to evolution send file to chatwoot using webhook in base64 formate, for this larger media file, the base64 code is too big. so chatwoot server is crashing.
You either need to talk to evolution API developer to support for media file instead of base64, or you need to hire any expert in this field, who can fix this issue.

**Comment by shivamkb17** [comment]
I am here responding, due to I faced similar problem while testing the unofficial API integration. but I am using wppconnect that time.

**Comment by BrOrlandi** [comment]
Great find @shivamkb17, so I think it needs to be fixed on both Chatwoot and Evolution.

On Chatwoot, it's possible to avoid processing the base64 when it's too big? Maybe it's important to have a notification in the chat, so the Agent can be notified the user tried to send a big file that can't be processed.

On Evolution, it need to send big files through media file instead of base64.

I will talk to the Evolution team about it so they may change to use the media file instead of base64.

**Comment by pranavrajs** [comment]
@BrOrlandi @shivamkb17 We'd be happy to accept a PR to limit the document size on the PR if you are interested.

**Comment by shivamkb17** [comment]
@pranavrajs Sure I will take a look into it. What can I do.

**Comment by BrOrlandi** [comment]
Hi guys, any updates on this? This error still occurs on Chatwot v3.16.0. Also, it's not related to big files, only to PSD files.
I have a 10mb PSD file that caused the crash, it's inside the zip attached. I'm updating the issue title as it's not related to 200mb file.
I think it may be related to the file encoding, this file I have is encoded on **windows-1252**.

@shivamkb17 can you try to reproduce with this file?

[Gabarito-photoshop_Embalagem-L261xA9xP18cm.psd.zip](https://github.com/user-attachments/files/19130379/Gabarito-photoshop_Embalagem-L261xA9xP18cm.psd.zip)

**Comment by BrOrlandi** [comment]
Both of the messages were not received even the second one is confirmed on chatwoot.
During the send of these files the CPU have peaks on sidekick container of 100% usage. I test this on a 24gb RAM 4 CPU server.
![Image](https://github.com/user-attachments/assets/6eb5c594-1157-4401-8a10-647e441c1f8e)

**Comment by BrOrlandi** [comment]
I reencoded the file as UTF-8, but it still doesn't work, even though it has confirmation, it was not sent to the API inbox.

Here is the file in the attachment to reproduce too.

[Gabarito-photoshop_Embalagem-L261xA9xP18cm (1) utf-8.psd.zip](https://github.com/user-attachments/files/19131506/Gabarito-photoshop_Embalagem-L261xA9xP18cm.1.utf-8.psd.zip)

![Image](https://github.com/user-attachments/assets/9cce866b-9850-4603-9419-69d566efa098)

**Comment by BrOrlandi** [comment]
Anyone could reproduce this issue? @shivamkb17
